backend/app/main.py

```
@@ -479,153 +461,84 @@ async def get_mcp_logs():
+        # Run discovery in the background
+        # Pass crawl_jobs dictionary to the background task if direct import causes issues
+        background_tasks.add_task(discover_pages, url=root_url, max_depth=request.depth, root_url=root_url, job_id=job_id)
+
+        logger.info(f"Discovery initiated in background for job ID: {job_id}")
+        # Return job ID immediately
+        return {
+            "message": "Discovery initiated",
+            "job_id": job_id,
             "success": True
         }
         logger.info(f"Returning response with {len(response_data['pages'])} pages")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)